data-prepper-plugins/key-value-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessorTests.java

```diff
@@ -315,6 +316,111 @@ void testIncludeExcludeKeysOverlapKeyValueProcessor() {
+
+        final Record<Event> record = getMessage("key1=value1&key2=value2");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) keyValueProcessor.doExecute(Collections.singletonList(record));
+        final LinkedHashMap<String, Object> parsed_message = getLinkedHashMap(editedRecords);
+
+        assertThat(parsed_message.size(), equalTo(1));
+        assertThatKeyEquals(parsed_message, "key1", "value1");
+    }
+
+    @Test
+    void testDefaultPrioritizeIncludeKeysKeyValueProcessor() {
+        final Map<String, Object> defaultMap = Map.of("key2", "value2");
+        final List<String> includeKeys = List.of("key1");
+        when(mockConfig.getDefaultValues()).thenReturn(defaultMap);
+        when(mockConfig.getIncludeKeys()).thenReturn(includeKeys);
+        keyValueProcessor = new KeyValueProcessor(pluginMetrics, mockConfig);
+
+        final Record<Event> record = getMessage("key1=value1&key2=abc");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) keyValueProcessor.doExecute(Collections.singletonList(record));
+        final LinkedHashMap<String, Object> parsed_message = getLinkedHashMap(editedRecords);
+
+        assertThat(parsed_message.size(), equalTo(2));
+        assertThatKeyEquals(parsed_message, "key1", "value1");
+        assertThatKeyEquals(parsed_message, "key2", "value2");
+    }
+
+    @Test
+    void testIncludeKeysNotInRecordMessageKeyValueProcessor() {
+        final Map<String, Object> defaultMap = Map.of("key2", "value2");
+        final List<String> includeKeys = List.of("key1");
+        when(mockConfig.getDefaultValues()).thenReturn(defaultMap);
+        when(mockConfig.getIncludeKeys()).thenReturn(includeKeys);
+        keyValueProcessor = new KeyValueProcessor(pluginMetrics, mockConfig);
+
+        final Record<Event> record = getMessage("key2=abc");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) keyValueProcessor.doExecute(Collections.singletonList(record));
+        final LinkedHashMap<String, Object> parsed_message = getLinkedHashMap(editedRecords);
+
+        assertThat(parsed_message.size(), equalTo(1));
+        assertThatKeyEquals(parsed_message, "key2", "value2");
+    }
+
+    @Test
+    void testDefaultExcludeKeysOverlapKeyValueProcessor() {
+        final Map<String, Object> defaultMap = Map.of("dKey", "dValue");
+        final List<String> excludeKeys = List.of("dKey");
+        when(mockConfig.getDefaultValues()).thenReturn(defaultMap);
+        when(mockConfig.getExcludeKeys()).thenReturn(excludeKeys);
+
+        assertThrows(IllegalArgumentException.class, () -> new KeyValueProcessor(pluginMetrics, mockConfig));
+    }
+
+    @Test
+    void testDefaultDuplicateKeysKeyValueProcessor() {
+        final Map<String, Object> defaultMap = Map.of("key2", "value2");
+        when(mockConfig.getDefaultValues()).thenReturn(defaultMap);
+        when(mockConfig.getSkipDuplicateValues()).thenReturn(true);
```

Comment:
It should be set to false to allow duplicate value and verify under that circumstance the default key-value will still not be involved.

Reply:
Actually by default we set skip_duplicate_values to false, so it may be redundant with other test case.

Reply:
I still think it may be a good test case to include in case anyone in the future is wondering about this specific case

Reply:
oh actually I see what you are saying, where if we set `skip_duplicate_values ` to `true` there's not really a point in having the test. will remove

Reply:
The other tests already test `skip_duplicate_values` set to false as that is the default.

I do think we should bring back a test to include `skip_duplicate_values` set to true because that could result in different behavior.

Actually, because `skip_duplicate_values` doesn't really affect default values, you could use a parameterized test on some of the existing tests. That is, regardless of `skip_duplicate_values`, finding the value in the Event should override the default.